en/24/10.md

```diff
@@ -0,0 +1,356 @@
+            }
+
+            #[only_owner]
+            #[endpoint]
+            fn withdraw(&self) {
+            let caller_address = self.blockchain().get_caller();
+            let collected_fees = self.collected_fees().get();
+            self.send().direct_egld(&caller_address, &collected_fees);
+            self.collected_fees().clear();
+            }
+          }
+    answer: >
+      multiversx_sc::imports!();
+
+      use crate::{storage, zombie::Zombie, zombiefactory, zombiefeeding, zombiehelper};
+
+      #[multiversx_sc::module]
+      pub trait ZombieAttack:
+          storage::Storage + zombie_feeding::ZombieFeeding + zombie_factory::ZombieFactory + zombie_helper::ZombieHelper
+      {
+          fn rand_mod(&self, modulus: usize) -> usize {
+              let mut rand_source = RandomnessSource::new();
+              rand_source.next_usize() % modulus
+          }
+
+          #[endpoint]
+          fn attack(&self, zombie_id: usize, target_id: usize) {
+              let caller = self.blockchain().get_caller();
+              self.check_zombie_belongs_to_caller(zombie_id, &caller);
+              let rand = self.rand_mod(100u8);
+              let attack_victory_probability = self.attack_victory_probability().get();
+              if rand <= attack_victory_probability {
+                  self.zombies(&zombie_id).update(|my_zombie| {
+                      my_zombie.win_count += 1;
+                      my_zombie.level += 1;
+                  });
+
+                  let mut enemy_dna = 0;
+                  self.zombies(&target_id).update(|enemy_zombie| {
+                      enemy_zombie.loss_count += 1;
+                      enemy_dna = enemy_zombie.dna;
+                  });
+                  self.feed_and_multiply(zombie_id, enemy_dna, ManagedBuffer::from(b"zombie"));
+              } 
+          }
+      }
+---
+
+Now that we have a `win_count` and `loss_count`, we can update them depending on which zombie wins the fight.
+
+In chapter 6 we calculated a random number from 0 to 100. Now let's use that number to determine who wins the fight, and update our stats accordingly.
+
+## Put it to the test
+
+1. Create an `if` statement that checks if `rand` is **_less than or equal to_** `attack_victory_probability`.
+
+2. If this condition is true, our zombie wins! So:
+
+  b. Declare a variable named `enemy_zombie`, and set it equal to `self.zombies(&target_id).get()`.
+  a. Access `self.zombies(&zombie_id)` udating it by increasing `win_count` and `level` by 1. (Level up!!!!!!!)
+
+  c. Accees `self.zombies(&target_id)` `loss_count`. (Loser!!!!!! 😫 😫 😫)
+
+  d. Run the `feed_and_multiply` function. Check `zombie_feeding.rs` to see the syntax for calling it. For the 3rd argument (`species`), pass a ManagedBuffer from `"zombie"` as bytes. (It doesn't actually do anything at the moment, but later we could add extra functionality for spawning zombie-based zombies if we wanted to).
```

Comment:
Does this mean they have to cast the parameter? if so, please be more specific and consider giving an example here or referring to where this is explained.